src/lib/message/L1ToL2Message.ts

```diff
@@ -345,12 +349,32 @@ export class L1ToL2MessageReader extends L1ToL2Message {
         if (successfulRedeem.length == 1) return successfulRedeem[0]
 
         const toBlock = await this.l2Provider.getBlock(toBlockNumber)
-        // dont bother looking past the retryable lifetime of the ticket for now
-        if (
-          toBlock.timestamp - creationBlock.timestamp >
-          l2Network.retryableLifetimeSeconds
-        )
-          break
+        if (toBlock.timestamp > timeout) {
+          // Check for LifetimeExtended event
+          while (queriedRange.length > 0) {
+            const blockRange = queriedRange.shift()
+            const keepaliveEvents = await eventFetcher.getEvents(
+              ARB_RETRYABLE_TX_ADDRESS,
+              ArbRetryableTx__factory,
+              contract =>
+                contract.filters.LifetimeExtended(this.retryableCreationId),
+              {
+                fromBlock: blockRange!.from,
+                toBlock: blockRange!.to,
+              }
+            )
+            if (keepaliveEvents.length > 0) {
+              timeout = keepaliveEvents
+                .map(e => e.event.newTimeout.toNumber())
+                .sort()
+                .reverse()[0]
+              break
+            }
+          }
+          if (toBlock.timestamp > timeout) break
+          // It is possible to have another keepalive in the last range as it might include block after previous timeout
+          while (queriedRange.length > 1) queriedRange.shift()
+        }
         const processedSeconds = toBlock.timestamp - fromBlock.timestamp
         if (processedSeconds != 0) {
           // find the increment that cover ~ 1 day
```

Comment:
I think we should do:
```
increment = Math.ceil(increment * 86400 / processedSeconds)
```
instead of:
```
increment *= Math.ceil(86400 / processedSeconds)
```
since we're trying to approach the target of processedSeconds == 86400 the value of `Math.ceil(86400 / processedSeconds)` should always be either 1 or 2, which is quite a large jump. Rounding after multiplication will yield smaller increments.

Reply:
nice catch, pushed the fix